youtube_dl/extractor/jwplatform.py

```diff
@@ -50,7 +50,11 @@ def _parse_jwplayer_data(self, jwplayer_data, video_id=None, require_title=True,
             # JWPlayer backward compatibility: flattened sources
             # https://github.com/jwplayer/jwplayer/blob/v7.4.3/src/js/playlist/item.js#L29-L35
             if 'sources' not in video_data:
-                video_data['sources'] = [video_data]
+                if isinstance(video_data, list):
```

Comment:
> That said, how would you suggest fixing it in iqm2.py without substantial code duplication?

Oops, for some reason I was thinking I'd need to override `_parse_jwplayer_data()` which is fairly long, but I shoehorned it into an override of `_extract_jwplayer_data()` instead. So done.

It still seems to me that this change would have benefited `jwplayform.py` and moving it into IQM2 is not advantageous, but up to you.